tests/functional/settings_secondary_emails.js

```diff
@@ -60,6 +60,59 @@ define([
       return this.remote.then(clearBrowserState());
     },
 
+    'gated in unverified session open verification same tab': function () {
+      return this.remote
+        // when an account is created, the original session is verified
+        // re-login to destroy original session and created an unverified one
+        .then(openPage(SIGNUP_URL, selectors.SIGNUP.HEADER))
```

Comment:
Can you simplify these tests by manually creating a user and then signing in? The first time the user signs in, will the session will be unverified?

```js
.then(createUser(email, PASSWORD, { preVerified: true }))
.then(openPage(SIGNIN_URL, selectors.SIGNIN.HEADER))
.then(fillOutSignIn(email, PASSWORD))
.then(testElementExists(selectors.UNLOCK.UNLOCK_BUTTON))
...
```